File: 品味Java的21種設計模式/dp_cpp/command/example1/ConcreteCommand.cpp
#include "ConcreteCommand.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace command
			{
				namespace example1
				{

					ConcreteCommand::ConcreteCommand(Receiver *receiver)
					{
						this->receiver = receiver;
					}

					void ConcreteCommand::execute()
					{
						//通常会转调接受者对象的相应方法，让接受者来真正执行功能
						receiver->action();
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/command/example1/ConcreteCommand.java
package cn.javass.dp.command.example1;
/**
 * 具体的命令实现对象
 */
public class ConcreteCommand implements Command {
    /**
     * 持有相应的接受者对象
     */
    private Receiver receiver = null;
    /**
     * 示意，命令对象可以有自己的状态
     */
    private String state;
    /**
     * 构造方法，传入相应的接受者对象
     * @param receiver 相应的接受者对象
     */
    public ConcreteCommand(Receiver receiver) {
        this.receiver = receiver;
    }

    public void execute() {
        //通常会转调接受者对象的相应方法，让接受者来真正执行功能
        receiver.action();
    }

}
